src/main/java/apoc/atomic/Atomic.java

```diff
@@ -0,0 +1,154 @@
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		String oldValue = propertyContainer.getProperty(property).toString();
+		String newValue = oldValue.concat(string);
+		propertyContainer.setProperty(property,newValue);
+		return Stream.of(new AtomicResults(oldValue,newValue));
+	}
+
+	/**
+	 * insert a value into an array property value
+	 */
+	@Procedure(mode = Mode.WRITE)
+	@Description("apoc.atomic.insert(node/relatonship,propertyName,position,value) insert a value into the property's array value at 'position'")
+	public Stream<AtomicResults> insert(@Name("container") Object container, @Name("propertyName") String property, @Name("position") Long position, @Name("value") Object value) throws ClassNotFoundException {
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		Object oldValue = propertyContainer.getProperty(property);
+		List<Object> values = insertValueIntoArray(oldValue, position, value);
+		Class clazz = Class.forName(values.toArray()[0].getClass().getName());
+		Object newValue = Array.newInstance(clazz, values.size());
+		try {
+			System.arraycopy(values.toArray(), 0, newValue, 0, values.size());
+		} catch (Exception e) {
+			String message = "Property's array value has type: " + values.toArray()[0].getClass().getName() + ", and your value to insert has type: " + value.getClass().getName();
+			throw new ArrayStoreException(message);
+		}
+		propertyContainer.setProperty(property, newValue);
+		return Stream.of(new AtomicResults(oldValue,newValue));
+	}
+
+	/**
+	 * remove a value into an array property value
+	 */
+	@Procedure(mode = Mode.WRITE)
+	@Description("apoc.atomic.remove(node/relatonship,propertyName,position) remove the element at position 'position'")
+	public Stream<AtomicResults> remove(@Name("container") Object container, @Name("propertyName") String property, @Name("position") Long position)
+			throws ClassNotFoundException {
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		Object oldValue = propertyContainer.getProperty(property);
+		List<?> values = new ArrayList<>();
+		values.addAll(new ArrayBackedList(oldValue));
+		Class clazz = Class.forName(values.toArray()[0].getClass().getName());
+		values.remove(position.intValue());
+		Object newValue = Array.newInstance(clazz, values.size());
+		System.arraycopy(values.toArray(), 0, newValue, 0, values.size());
+		propertyContainer.setProperty(property, newValue);
+		return Stream.of(new AtomicResults(oldValue,newValue));
+	}
+
+	/**
+	 * update the property's value
+	 */
+	@Procedure(mode = Mode.WRITE)
+	@Description("apoc.atomic.update(node/relatonship,propertyName,updateOperation) update a property's value with a cypher operation (ex. \"n.prop1+n.prop2\")")
+	public Stream<AtomicResults> update(@Name("container") Object container, @Name("propertyName") String property, @Name("operation") String operation){
+		PropertyContainer propertyContainer = (PropertyContainer) container;
```

Comment:
we have to take a lock again.
not sure why you used iterate here, for a singular update.